src/main/java/org/opensearch/knn/index/query/KNNWeight.java

```diff
@@ -296,10 +324,12 @@ private Map<Integer, Float> doANNSearch(
             }
             int[] parentIds = getParentIdsArray(context);
             if (k > 0) {
-                if (knnQuery.getVectorDataType() == VectorDataType.BINARY) {
+                if (knnQuery.getVectorDataType() == VectorDataType.BINARY
+                    || quantizationParams != null && quantizationService.getVectorDataTypeForTransfer(fieldInfo) == VectorDataType.BINARY) {
```

Comment:
Instead of this create a variable `finalVectorDataType`/ `processedVectorDataType` and compute the vector data and then use it everywhere

```
finalVectorDataType = knnQuery or quantizedService as a source

```

Reply:
As of right now, we only use this check here so it would only be used once.  Everywhere else it just reads from the query vector type and does not need the influence of the quantization service.